test/explain/algorithm/test_gnn_explainer.py

```diff
@@ -0,0 +1,389 @@
+        assert explanation.node_features_mask.shape == x.shape
+        assert explanation.node_features_mask.min() >= 0
+        assert explanation.node_features_mask.max() <= 1
+        assert explanation.node_features_mask[
+            0] == explanation.node_features_mask[1]
+
+    if edge_mask_type == MaskType.object:
+        assert explanation.edge_mask.shape == (edge_index.shape[1], )
+        assert explanation.edge_mask.min() >= 0
+        assert explanation.edge_mask.max() <= 1
+
+
+# type of masks allowed for GNNExplainer
+node_mask_types = ["attributes", "object", "common_attributes"]
+edge_mask_types = ["object", None]
+return_types_classification = ["log_probs", "raw", "probs"]
+return_types_regression = ["raw"]
+
+x = torch.randn(8, 3)
+edge_index = torch.tensor([[0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7],
+                           [1, 0, 2, 1, 3, 2, 4, 3, 5, 4, 6, 5, 7, 6]])
+
+
+@pytest.mark.parametrize('edge_mask_type', edge_mask_types)
+@pytest.mark.parametrize('node_mask_type', node_mask_types)
+@pytest.mark.parametrize('model', [
+    GNN_regression_singleoutput,
+    GNN_regression_multioutput,
+])
+@pytest.mark.parametrize('return_type', return_types_regression)
+@pytest.mark.parametrize("explanation_type", ["model", "phenomenon"])
+def test_gnn_explainer_with_meta_explainer_regression_graph(
+    edge_mask_type,
+    node_mask_type,
+    model,
+    return_type,
+    explanation_type,
+):
+    explainer_config = ExplainerConfig(
+        explanation_type=explanation_type,
+        node_mask_type=node_mask_type,
+        edge_mask_type=edge_mask_type,
+    )
+    model_config = ModelConfig(
+        task_level='graph',
+        return_type=return_type,
+        mode="regression",
+    )
+
+    model = model()
+    model.eval()
+    with torch.no_grad():
+        out = model(x, edge_index, None, None)
+
+    if isinstance(
+            model,
+            GNN_regression_multioutput,
+    ):
+        target_index = 0
+    else:
+        target_index = None
+
+    explainer = Explainer(model=model, algorithm=GNNExplainer(epochs=2),
+                          explainer_config=explainer_config,
+                          model_config=model_config)
+
+    # try to explain prediction for node 0
+    explanation = explainer(x=x, edge_index=edge_index, target=out,
+                            target_index=target_index, edge_attr=None,
+                            batch=None)
+
+    check_explanation(edge_mask_type, node_mask_type, x, edge_index,
+                      explanation)
+
+
+@pytest.mark.parametrize('edge_mask_type', edge_mask_types)
+@pytest.mark.parametrize('node_mask_type', node_mask_types)
+@pytest.mark.parametrize(
+    'model', [GNN_classification_singleoutput, GNN_classification_multioutput])
+@pytest.mark.parametrize("explanation_type", ["model", "phenomenon"])
+@pytest.mark.parametrize('return_type', return_types_regression)
+def test_gnn_explainer_with_meta_explainer_classification_graph(
```

Comment:
Small, optional request: Could you put the node tests before the graph tests since the node models are defined first. Also pick an ordering between classification/regression and single/multi output and stay consistent across model definitions and tests. 

Not sure if this is truly necessary but helps with readability for me.